app/src/main/java/com/github/se/studybuddies/ui/groups/GroupMembers.kt

```diff
@@ -0,0 +1,291 @@
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import coil.compose.rememberAsyncImagePainter
+import com.github.se.studybuddies.R
+import com.github.se.studybuddies.data.User
+import com.github.se.studybuddies.database.DbRepository
+import com.github.se.studybuddies.navigation.GROUPS_MEMBERS_DESTINATIONS
+import com.github.se.studybuddies.navigation.NavigationActions
+import com.github.se.studybuddies.navigation.Route
+import com.github.se.studybuddies.ui.shared_elements.GoBackRouteButton
+import com.github.se.studybuddies.ui.shared_elements.Sub_title
+import com.github.se.studybuddies.ui.shared_elements.TopNavigationBar
+import com.github.se.studybuddies.ui.theme.Blue
+import com.github.se.studybuddies.ui.theme.White
+import com.github.se.studybuddies.viewModels.GroupViewModel
+
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter", "MutableCollectionMutableState")
+@Composable
+fun GroupMembers(
+    groupUID: String,
+    groupViewModel: GroupViewModel,
+    navigationActions: NavigationActions,
+    db: DbRepository
+) {
+
+  if (groupUID.isEmpty()) return
+  groupViewModel.fetchGroupData(groupUID)
+  val groupData by groupViewModel.group.observeAsState()
+
+  val nameState = remember { mutableStateOf(groupData?.name ?: "") }
+  val members = remember { groupData?.let { mutableStateOf(it.members) } }
+
+  val currUser = groupViewModel.getCurrentUser()
+
+  var nbMember = 0
+  groupData?.let {
+    nameState.value = it.name
+    if (members != null) {
+      members.value = it.members
+      nbMember = members.value.size
+    }
+  }
+
+  val userDatas = remember { mutableStateOf(mutableListOf<User>()) }
+
+  if (members != null) {
+    for (i in 0 until nbMember) {
+      Log.d("memberCheck", "$nbMember")
+      groupViewModel.fetchUserData(members.value[i])
+      val userData by groupViewModel.member.observeAsState()
+      if (userData?.username?.isNotBlank() == true) {
+        userData?.let { userDatas.value.add(it) }
+      }
+    }
+  }
```

Comment:
During testing, I got flooded by log. The fetchUserData was called each few ms.
```
2024-05-24 15:27:34.338 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.349 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.349 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.361 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.361 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.372 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.372 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.383 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.383 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.395 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.395 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.410 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.410 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.423 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.423 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.433 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.433 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.443 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.443 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.450 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.450 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.461 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.461 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.472 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.472 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.483 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.483 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.494 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.494 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.506 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.506 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.527 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.527 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.539 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.539 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.550 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.550 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.562 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.562 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.572 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.572 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.584 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.584 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.595 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.595 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.606 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.606 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.617 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.617 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.633 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.633 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.645 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.645 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.653 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.653 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.662 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.662 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.673 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.673 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.684 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.684 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.695 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.695 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.706 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.706 12827-12827 memberCheck             com.github.se.studybuddies           D  1
2024-05-24 15:27:34.718 12827-12827 MyPrint                 com.github.se.studybuddies           D  Fetched user UID is qMw4wwydC9S6riQHwcnJpoVHTsv1
2024-05-24 15:27:34.718 12827-12827 memberCheck             com.github.se.studybuddies           D  1

```

Reply:
It work exactly like the other function we have like for groupHome, I think it's normal

Reply:
But I think that groupHome as an issue (that caused our quota limit to exceed), however if you said it's normal, I will accept and we will need to figure it out later